src/structure/model.jl

```diff
@@ -9,211 +9,288 @@ export
     het_res::Bool # Does the residue consist of hetatoms?
     atom_list::Vector{String}
     atoms::Dict{String, AbstractAtom}
+    chain::StructuralElement
 end
 
-# Constructor without atoms
-Residue(name::AbstractString, chain_id::Char, number::Integer, ins_code::Char, het_res::Bool) = Residue(
-    name, chain_id, number, ins_code, het_res, [], Dict())
-# Constructor from a list of atoms
-Residue{T <: AbstractAtom}(atoms::Vector{T}) = Residue(
-        resname(atoms[1]),
-        chainid(atoms[1]),
-        resnumber(atoms[1]),
-        inscode(atoms[1]),
-        ishetero(atoms[1]),
-        map(atomname, sort(atoms)),
-        Dict(atomname(atom) => atom for atom in atoms))
-
 "A container to hold different versions of the same residue (point mutations)."
 immutable DisorderedResidue <: AbstractResidue
     names::Dict{String, Residue}
     default::String
 end
 
 
-"A chain represented in a PDB file."
+"A chain (molecule) from a macromolecular structure."
 immutable Chain <: StructuralElement
     id::Char
     res_list::Vector{String}
     residues::Dict{String, AbstractResidue}
+    model::StructuralElement
 end
 
-Chain(id::Char) = Chain(id, [], Dict())
 
-
-"A conformation of a protein represented in a PDB file."
+"A conformation of a macromolecular structure."
 immutable Model <: StructuralElement
     number::Int
     chains::Dict{Char, Chain}
+    structure::StructuralElement
 end
 
-Model(number::Integer) = Model(number, Dict())
-Model() = Model(1)
-
 
-"A container for multiple models from a PDB file."
+"A container for multiple `Model`s."
 immutable ProteinStructure <: StructuralElement
     name::String
     models::Dict{Int, Model}
 end
 
+
+"""
+A record for a single atom, e.g. as represented in a Protein Data Bank (PDB)
+file.
+"""
+immutable AtomRecord
+    het_atom::Bool
+    serial::Int
+    atom_name::String
+    alt_loc_id::Char
+    res_name::String
+    chain_id::Char
+    res_number::Int
+    ins_code::Char
+    coords::Vector{Float64}
+    occupancy::Float64
+    temp_fac::Float64
+    element::String
+    charge::String
+end
+
+
+# Constructors without sub-elements and without super-elements
+
 ProteinStructure(name::AbstractString) = ProteinStructure(name, Dict())
+
 ProteinStructure() = ProteinStructure("")
 
 
+Model(number::Integer, struc::ProteinStructure) = Model(number, Dict(), struc)
+
+Model(struc::ProteinStructure) = Model(1, struc)
+
+Model(number::Integer) = Model(number, ProteinStructure())
+
+Model() = Model(ProteinStructure())
+
+
+Chain(id::Char, mod::Model) = Chain(id, [], Dict(), mod)
+
+Chain(mod::Model) = Chain('A', mod)
+
+Chain(id::Char) = Chain(id, Model())
+
+Chain() = Chain('A')
+
+
+function Residue(name::AbstractString,
+                number::Integer,
+                ins_code::Char,
+                het_res::Bool,
+                ch::Chain)
+    return Residue(name, number, ins_code, het_res, [], Dict(), ch)
+end
+
+Residue(ch::Chain) = Residue("ALA", 1, ' ', false, ch)
```

Comment:
I think this kind of default values (`"ALA"`, `1`) will lead to hard-to-find bugs in the future. Is that possible to force making a structure from outer (`ProteinSructure`) to inner (`Atom`) elements? Passing a wrapper element to constructors will remove the necessity of these default values.